vyper/builtin_functions/functions.py

```diff
@@ -526,10 +555,10 @@ def build_IR(self, expr, context):
             ]
         )
 
-        if current_type == "String":
-            ret_typ = StringType(maxlen=dst_maxlen)
-        else:
+        if isinstance(args[0].typ, ByteArrayType) or is_bytes_m_type(args[0].typ):
```

Comment:
nit: condition would be simpler if the branches were flipped
```suggestion
        if isinstance(args[0].typ, StringType):
```